### DISCUSSION TITLE
Disable recent visited Sites

### DISCUSSION THREAD

**Discussion by iamthelara** [bromite/bromite#1224] [Ideas]
hey, is there any way to disable recent visited sites?
It's causing mess in my home page.

**Comment by ArigatoDevs** [comment]
Hi. What you can try doing is selecting a home page of website of your choice. Go to settings, then Homepage, then choose any website of your liking to be the home page.

**Reply by iamthelara** [reply]
No, I want chrome home page, but i want to disable recent visited websites on homepage.

**Reply by csagan5** [reply]
That home page would always stay empty.

**Reply by iamthelara** [reply]
yeah, i know. That would be fine. 
and if we will able to customize the homepage with desired sites as short cut it would be great.

**Reply by csagan5** [reply]
Disabling `#site-engagement` will disable automatically-created shortcuts, but I do not know whether it is possible to manually put there bookmarks.

**Comment by Bugaddr** [comment]
Is there any config or something else to disable this (history on homepage) ?

**Reply by csagan5** [reply]
Disable the `#site-engagement` flag, that should do the job.

**Reply by Bugaddr** [reply]
Thanks csagan5.

**Reply by Issam28** [reply]
> Disable the `#site-engagement` flag, that should do the job.

I already disabled this flag, but I'm still seeing icons of my most recent websites, any idea why?

**Reply by Issam28** [reply]
As a follow up to my comment, I noticed that the most visited websites show up only when you access them from history. if you visit a website from history two or three times, an icon will show up in the NTP page, no idea why it is like this but hope this helps the dev to figure out a fix. Thanks!

**Reply by uazo** [reply]
@csagan5 I had seen this thing before, look here
https://github.com/uazo/bromite/blob/master-v101/build/patches/contrib/Remove-some-trace-on-disk.patch#L19
I haven't had time to investigate, but I think that's the point in the code

**Reply by Bugaddr** [reply]
Yeah, i too faced the issue again, sorry i forgot to report here as for some time you will not get any shortcut after disabling flag but it suddenly appear on next launch.

**Reply by csagan5** [reply]
> I had seen this thing before, look here
https://github.com/uazo/bromite/blob/master-v101/build/patches/contrib/Remove-some-trace-on-disk.patch#L19
I haven't had time to investigate, but I think that's the point in the code

@uazo I do not follow. Is this a bug that makes site engagement active even when disabled, or it is just another mechanism to add NTP icons (unrelated to site engagement)?

**Reply by uazo** [reply]
no, sorry, different things.

> I had seen this thing before ... I haven't had time to investigate

`visitedlink::VisitedLinkWriter` manages blink's `:visited` pseudo class, one thing that I would take away, because of [high-bandwidth leaking of :visited state](https://github.com/w3c/csswg-drafts/issues/3012)

> bug that makes site engagement active even when disabled

that patch turns off scoring, but i don't know if ntp uses that for the list.

**Reply by uazo** [reply]
checked, this is the source of ntp tile
https://source.chromium.org/chromium/chromium/src/+/main:components/history/core/browser/history_backend.cc;drc=b9be6b1f2c961debcffa012b514c88444ccb8fcc;bpv=1;bpt=1;l=2262

**Reply by csagan5** [reply]
Ok, it uses the history database. So there is currently no way to disable NTP tiles based on history without disabling history.

**Reply by uazo** [reply]
> So there is currently no way to disable NTP tiles based on history without disabling history.

well maybe yes, if the source `segment_usage` isn't used elsewhere

```
"SELECT segment_id, time_slot, visit_count "
"FROM segment_usage ORDER BY segment_id"
```

but it's worth it?

**Reply by csagan5** [reply]
I don't understand, what is worth it?

**Reply by uazo** [reply]
I intend to check if it is possible to delete the `segment_usage` table management without deleting all history management.
after all, if you don't want the tiles just put `about:blank` in the homepage settings
